core/src/kmx/kmx_plus.cpp

```diff
@@ -314,23 +347,19 @@ KMX_DWORD COMP_KMXPLUS_SECT::find(KMX_DWORD ident) const {
   return 0;
 }
 
-PKMX_WCHAR
-COMP_KMXPLUS_STRS::get(KMX_DWORD entry, PKMX_WCHAR buf, KMX_DWORD bufsiz) const {
+std::u16string
+COMP_KMXPLUS_STRS::get(KMX_DWORD entry) const {
     assert(entry < count);
     if (entry >= count) {
-        return nullptr;
+        return std::u16string(); // Fallback: empty string
     }
-    // TODO-LDML: will be improved in https://github.com/keymanapp/keyman/issues/7134
     KMX_DWORD offset = entries[entry].offset;
     KMX_DWORD length = entries[entry].length;
-    assert(bufsiz > (length+1)); // assert bufsiz big enough
     assert(offset+((length+1)*2) <= header.size); // assert not out of bounds
     const uint8_t* thisptr = reinterpret_cast<const uint8_t*>(this);
     const KMX_WCHAR* start = reinterpret_cast<const KMX_WCHAR*>(thisptr+offset);
-    for(KMX_DWORD i=0;i<=length;i++) {
-        buf[i] = start[i];
-    }
-    return buf;
+    assert(start[length] == 0); // assert null terminated
```

Comment:
We need to be sure that assertions are around bugs in our code, not malformed inputs. Not sure that this is covering that off?

Reply:
removed